### PR TITLE
RHMAP-10126 Migrate to Swift 3

### PR DESCRIPTION
**Pull request by jcesarmobile**

### CODE REVIEW

File: sync-ios-app.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -363,23 +363,23 @@
 				PRODUCT_BUNDLE_IDENTIFIER = "org.feedhenry.sync-ios-app";
 				PRODUCT_NAME = "$(TARGET_NAME)";
 				SWIFT_OPTIMIZATION_LEVEL = "-Onone";
-				SWIFT_VERSION = 2.3;
+				SWIFT_VERSION = 3.0;
```
Comment (corinnekrych) [inline sync-ios-app.xcodeproj/project.pbxproj]: let's use 3.0.1

File: sync-ios-app/AppDelegate.swift [hunk 2]
```diff
@@ -54,31 +54,31 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
         let coordinator = NSPersistentStoreCoordinator(managedObjectModel: self.managedObjectModel)
-        let url = self.applicationDocumentsDirectory.URLByAppendingPathComponent("ShoppingItem.sqlite")
+        let url = self.applicationDocumentsDirectory.appendingPathComponent("ShoppingItem.sqlite")
         var failureReason = "There was an error creating or loading the application's saved data."
         do {
-            try coordinator.addPersistentStoreWithType(NSSQLiteStoreType, configuration: nil, URL: url, options: nil)
+            try coordinator.addPersistentStore(ofType: NSSQLiteStoreType, configurationName: nil, at: url, options: nil)
         } catch {
             // Report any error we got.
             var dict = [String: AnyObject]()
```
Comment (corinnekrych) [inline sync-ios-app/AppDelegate.swift]: if you use [String: Any] you could avoid the casting: "a string" as AnyObject? 2 following lines

**Comment by jcesarmobile** [comment]
@corinnekrych I made the changes

**Comment by corinnekrych** [comment]
+1

**Pull request merged**